src/lib/deskflow/ProtocolTypes.h

```diff
@@ -11,365 +11,1355 @@
+ * This prevents memory exhaustion attacks.
+ *
+ * @note Clipboard messages are separately limited to 32KB chunks
+ * @since Protocol version 1.0
+ */
 static constexpr uint32_t PROTOCOL_MAX_MESSAGE_LENGTH = 4 * 1024 * 1024;
+
+/**
+ * @brief Maximum allowed list length in protocol messages
+ *
+ * Limits the size of lists (arrays) in protocol messages to prevent
+ * memory exhaustion attacks.
+ *
+ * @since Protocol version 1.0
+ */
 static constexpr uint32_t PROTOCOL_MAX_LIST_LENGTH = 1024 * 1024;
+
+/**
+ * @brief Maximum allowed string length in protocol messages
+ *
+ * Limits the size of strings in protocol messages to prevent
+ * memory exhaustion attacks.
+ *
+ * @since Protocol version 1.0
+ */
 static constexpr uint32_t PROTOCOL_MAX_STRING_LENGTH = 1024 * 1024;
 
-// direction constants
+/** @} */ // end of protocol_constants group
+
+/**
+ * @defgroup protocol_enums Protocol Enumerations
+ * @brief Enumeration types used in protocol messages
+ * @{
+ */
+
+/**
+ * @brief Screen edge directions for mouse movement
+ *
+ * Used to specify which edge of a screen the mouse cursor crosses
+ * when moving between primary and secondary screens.
+ *
+ * @since Protocol version 1.0
+ */
 enum EDirection
 {
-  kNoDirection,
-  kLeft,
-  kRight,
-  kTop,
-  kBottom,
-  kFirstDirection = kLeft,
-  kLastDirection = kBottom,
-  kNumDirections = kLastDirection - kFirstDirection + 1
+  kNoDirection,                                         ///< No specific direction
+  kLeft,                                                ///< Left edge of screen
+  kRight,                                               ///< Right edge of screen
+  kTop,                                                 ///< Top edge of screen
+  kBottom,                                              ///< Bottom edge of screen
+  kFirstDirection = kLeft,                              ///< First valid direction value
+  kLastDirection = kBottom,                             ///< Last valid direction value
+  kNumDirections = kLastDirection - kFirstDirection + 1 ///< Total number of directions
 };
+
+/**
+ * @brief Bitmask values for screen edge directions
+ *
+ * Used to create bitmasks representing multiple screen edges.
+ * Useful for configuration and edge detection.
+ *
+ * @since Protocol version 1.0
+ */
 enum EDirectionMask
 {
-  kNoDirMask = 0,
-  kLeftMask = 1 << kLeft,
-  kRightMask = 1 << kRight,
-  kTopMask = 1 << kTop,
-  kBottomMask = 1 << kBottom
+  kNoDirMask = 0,            ///< No direction mask
+  kLeftMask = 1 << kLeft,    ///< Left edge mask
+  kRightMask = 1 << kRight,  ///< Right edge mask
+  kTopMask = 1 << kTop,      ///< Top edge mask
+  kBottomMask = 1 << kBottom ///< Bottom edge mask
 };
 
-// Data transfer constants
+/**
+ * @brief File transfer data chunk types
+ *
+ * Used in kMsgDFileTransfer messages to indicate the type of data
+ * being transferred in each chunk.
+ *
+ * @since Protocol version 1.5
+ */
```

Comment:
I'll leave the wording up to you.
```suggestion
 * @since Protocol version 1.5
 * @deprecated
 */
```

Reply:
@sithlord48 Hmm, actually maybe the `EDataTransfer` and `EDataReceived` enums should be removed (in another PR)

Reply:
Yeah i can look at that .

Reply:
- https://github.com/deskflow/deskflow/issues/8713

added `@deprecated`

Reply:
restored